### ISSUE TITLE
Formation Flight

### ISSUE THREAD

**Issue by Avivps** [till213/SkyDolly#127]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**
Hi, if you can help? how to use the formation tumbails icon the small plan icons in the right corner at the page. its dosn't do nothing the relative position of the plane, maybe, I don't know hoe to use it sorry.
THANKS

Dylan

**Comment by till213** [comment]
Hello,

First of all, this relative position selector is just a crude version of some nicer user interface that should be in place here. So the current "Bearing" selector is just a quick and dirty solution.

In any case, it serves two things:

- Recording: It defines the relative position (bearing) when you start recording the next aircraft in the formation
- Replay: during replay it re-positions your user aircraft in the selected position

During recording make sure that the "Set relative position" is actually enabled: otherwise you start recording your new formation aircraft at its current location (this is useful when you e.g. want to record some "head vs head" flight, or want to start at the opposite end of some runway etc.)

The later option takes effect when you are actually flying yourself, together with the recorded formation flight. In order to do so you select the desired "Replay" mode, which is one of:

- Formation (Normal): all recorded aircraft are replayed (you are simply watching the replay)
- Take control of recorded user aircraft: you take the control of the user aircraft (marked with the blue aircraft icon in the list of formation aircraft) and fly along with the rest of the formation
- Fly with formation: all recorded formation aircraft are replayed, and you can follow them with your current user aicraft

In those last two replay options you can always select a new relative position and your current user aircraft will be repositioned next to the corresponding aircraft in the formation.

I hope this helps somewhat. And yes, the online manual should be updated accordingly ;)

**Comment by till213** [comment]
I will re-open this issue, as this really needs to be documented properly at some point in the online manual:

https://till213.github.io/SkyDolly/manual/en/

**Comment by HLE84** [comment]
I am trying something similar but when I fly with formation i am trown upside down and it breaks my drone cam. How do i start playing a file without having to get spawned in the formation?

**Comment by HLE84** [comment]
And do we know a way to get the time to stick with real time and not suddenly auto set the time as it wants?

**Comment by till213** [comment]
> How do i start playing a file without having to get spawned in the formation?

Yes, the online documentation... I did not touch it in a long time, shame on me. So for the time being I'll give you a quick answer here:

In the "Formation" module (in Sky Dolly) you have several options how (if) you want to "join" a formation flight while playing a flight with one or (typically) several recorded aircraft:

![grafik](https://github.com/user-attachments/assets/01d6128e-d895-4b2d-b52d-224caf0484e0)

- Formation (Normal)
  - All recorded aircraft - including your currently selected "user aircraft" - are controlled by Sky Dolly (= you do not have control over any aircraft)
- Take control of recorded user aircraft
  - You can control the recorded aircraft that is marked as "user aircraft" (a blue airplane icon appears next to the aircraft in the aircraft table - you can switch the "user aircraft" by selecting your desired aircraft and click the "Set User Aircraft" button)
  - Each time you start the replay you are automatically spawned ("teleported") to the recorded position of your selected user aircraft; but from that point onwards you have control
- Fly with formation
  - This is what could be considered "free flight": all recorded aircraft are controlled by Sky Dolly, but your currently selected "user aircraft" remains under your control - and flies _in addition_ to the recorded aircraft
  - Note that in this mode whenever you click one of the "relative position" airplane icons your "user aircraft" gets re-positioned next to the (former) "user aircraft" in the formation (now the "reference aircraft", marked with a green airplane icon)
  - Like this you can easily re-join the formation, should you get "lost"

Note that "user aircraft" means the aircraft that you have currently loaded in MSFS in your active flight.

So a concrete example: you want to import some flight path (e.g. from flightaware.com or flightradar24.com etc.) and join that flight with a second aircraft:

- Import the file into Sky Dolly (e.g. File | Flight Import | KML)
  - A new flight with the just imported aircraft is now active (loaded into memory, ready for replay)
- Switch to the "Formation" module (press key F2 or the corresponding button on top)
- Choose the "Fly with formation" formation replay option
- Press "Play"

Your current "user aircraft" now gets automatically re-positioned relative to the imported aircraft, but you now have immediate control and can "fly with the formation" (which in this case is the just imported aircraft).

Note that you can also import several KML files into the "same flight" and create an entire "airport scenario" - and join / fly with each aircraft of your choice.

If you do not wish do get re-positioned when pressing "Play" you can uncheck the corresponding "Set relative position" checkbox (this is especially useful when you record a formation flight and want to start recording from e.g. your current parking space, instead of relative to the last "formation user/reference aircraft").

I hope that makes sense somewhat.

**Comment by till213** [comment]
> And do we know a way to get the time to stick with real time and not suddenly auto set the time as it wants?

This is a somewhat unrelated question to "formation flying" (it is usually better to create a new issue for each topic at hand), but since it is a quick one:

![grafik](https://github.com/user-attachments/assets/66e5d14b-d860-4a73-a49a-5696108beb4c)

Simply select "None" (= no time synchronisation) in the Sky Dolly application settings.

This is a new feature that has been introduced with Sky Dolly v0.18 and has been documented in the [CHANGELOG](![grafik](https://github.com/user-attachments/assets/4c66a67b-5f1e-4f79-9ae0-3cb2ac802268)) - and also been mentioned in the Sky Dolly "splash screen" that appears 3 times for each new version (in case you are not reading those ;)).
